rust/lancedb/src/error.rs

```diff
@@ -49,6 +50,10 @@ pub enum Error {
     Arrow { source: ArrowError },
     #[snafu(display("LanceDBError: not supported: {message}"))]
     NotSupported { message: String },
+    #[cfg(feature = "polars")]
+    #[snafu(display("Polars error: {source}"))]
+    #[cfg(feature = "polars")]
+    Polars { source: polars::error::PolarsError },
```

Comment:
Let's go ahead and just make an `External` error type similar to the one in [`arrow-rs`](https://docs.rs/arrow-schema/51.0.0/arrow_schema/enum.ArrowError.html#variant.ExternalError) or [`datafusion`](https://docs.rs/datafusion/latest/datafusion/error/enum.DataFusionError.html#variant.External).  It will accept `Box<dyn Error + Send + Sync>` and so you don't need to hide it behind the polars feature gate.